File: fib04.go
package main

import (
	"fmt"
)

func fibonacci(n int) int {
	if n < 3 {
		return 1
	} else {
		return fibonacci(n-2) + fibonacci(n-1)
	}
}

func fibtest(n int) {
	fmt.Printf("%d : %d\n", n, fibonacci(n))
}

func main() {
	fibtest(44)
	fibtest(45)
	fibtest(46)
}

File: fib04.pas
program fibber;

function fibonacci(n: longint): longint;
begin
	if (n < 3) then
		fibonacci := 1
	else
		fibonacci := fibonacci(n-1) + fibonacci(n-2);
end;

procedure fibtest(n: longint);
begin
	writeln('fib(', n, ') = ', fibonacci(n))
end;

begin
	fibtest(44);
	fibtest(45);
	fibtest(46);
end.
